Sources/Graph/custom_graph.hpp

```diff
@@ -43,134 +39,23 @@ class CustomGraph : public AbstractGraph {
  public:
   CustomGraph(std::vector<Edge> edges, ColorMap colors = ColorMap(),
               std::vector<std::vector<int>> automorphisms =
-                  std::vector<std::vector<int>>())
-      : edges_{std::move(edges)},
-        eclist_{std::move(colors)},
-        automorphisms_{std::move(automorphisms)} {
-    n_sites_ = CheckEdges();
-    if (n_sites_ == 0) {
-      throw InvalidInputError{"Empty graphs are not supported."};
-    }
-    if (eclist_.empty() && !edges_.empty()) {
-      for (auto const &edge : edges_) {
-        eclist_.emplace(edge, 0);
-      }
-    } else {
-      CheckEdgeColors();
-    }
-    if (!automorphisms_.empty()) {
-      CheckAutomorph();
-    } else {
-      automorphisms_.resize(1);
-      automorphisms_.front().resize(static_cast<std::size_t>(n_sites_));
-      std::iota(std::begin(automorphisms_.front()),
-                std::end(automorphisms_.front()), 0);
-    }
-    is_connected_ = IsConnected();
-    is_bipartite_ = IsBipartite();
-  }
+                  std::vector<std::vector<int>>());
 
+  virtual int Nsites() const noexcept override;
```

Comment:
my ignorance: why do we need to declare it virtual here?

Reply:
`virtual` is redundant here (it is implied by `Nsites` overriding the base class method); I'd suggest we remove it.

Reply:
It's good practice to declare a function virtual when it is. And not just in the base class which is obvious, but also in derived classes. This helps avoid the confusion of overriding vs shadowing.